src/txmempool.cpp

```diff
@@ -221,7 +221,7 @@ void CTxMemPool::UpdateAncestorsOf(bool add, txiter it, setEntries &setAncestors
         UpdateChild(piter, it, add);
     }
     const int64_t updateCount = (add ? 1 : -1);
-    const int64_t updateSize = updateCount * it->GetTxSize();
+    const int64_t updateSize = updateCount * (int64_t)it->GetTxSize();
```

Comment:
Why is this necessary?

Reply:
Consider the case where `add == false` and `it->GetTxSize() == 2`.

Then an unsigned integer overflow occurs when calculating `updateCount * it->GetTxSize()` (but not for `updateCount * (int64_t)it->GetTxSize()`).

Reply:
:+1: